comms/dht/src/peer_validator.rs

```diff
@@ -67,55 +67,55 @@ impl<'a> PeerValidator<'a> {
-                        .as_ref()
-                        .map(|i| i.updated_at() < new_dt && (
-                            !current_peer.is_seed() ||
-                            current_peer.added_at < new_dt.naive_utc()))
-                        // If None, update to peer with valid signature
-                        .unwrap_or(true)
-                };
-
-                if !can_update {
-                    debug!(
-                        target: LOG_TARGET,
-                        "Peer `{}` already exists or is up to date and will not be updated", new_peer.node_id
-                    );
-                    return Ok(false);
-                }
+            Some(current_peer) => {
+                // let can_update = can_update && {
+                //     // Update/insert peer if newer
+                //     // unreachable panic: can_update is true only is identity_signature is present and valid
+                //     // let new_dt = new_peer
+                //     //     .identity_signature
+                //     //     .as_ref()
+                //     //     .map(|i| i.updated_at())
+                //     //     .expect("unreachable panic");
+                //
+                //     // Update if new_peer has newer timestamp than current_peer, and if the newer timestamp is after
+                // the     // added date
+                //     current_peer
+                //         .identity_signature
+                //         .as_ref()
+                //         .map(|i| i.updated_at() < new_dt && (
+                //             !current_peer.is_seed() ||
+                //             current_peer.added_at < new_dt.naive_utc()))
+                //         // If None, update to peer with valid signature
+                //         .unwrap_or(true)
+                // };
+
+                // if !can_update {
+                //     debug!(
+                //         target: LOG_TARGET,
+                //         "Peer `{}` already exists or is up to date and will not be updated", new_peer.node_id
+                //     );
+                //     return Ok(false);
+                // }
 
                 debug!(target: LOG_TARGET, "Updating peer `{}`", new_peer.node_id);
-                current_peer
-                    .update_addresses(new_peer.addresses.into_vec())
-                    .set_features(new_peer.features)
-                    .set_offline(false);
-                if let Some(sig) = new_peer.identity_signature {
-                    current_peer.set_valid_identity_signature(sig);
-                }
+                // current_peer
+                //     .update_addresses(new_peer.addresses.into_vec())
+                //     .set_features(new_peer.features);
+                // if let Some(sig) = new_peer.identity_signature {
+                //     current_peer.set_valid_identity_signature(sig);
+                // }
                 self.peer_manager.add_peer(current_peer).await?;
```

Comment:
Noting that add_peer is changed to only append new addresses to mitigate the fact that addresses are no longer signed. It also merges stats for addresses however that does not apply here because the peer sync message cannot and should not contain this data. 

Malicious case:
Should a remote peer F be allowed to add any amount (1000s, 10_000, ?) of addresses for any peer? If not, perhaps we should make the maximum e.g. 10. If the remote peer F provides 10 new addresses for peer A which already has 5 in our peer manager, we have no choice but to discard 5 of the addresses (at random?) provided by peer F. If peer A already has 10, we cannot add any addresses as all of them could be false. 

Comparison of happy path case:

Setup: Peer A has an updated address set, peer B is made aware of the change via an existing authenticated mechanism (DHT join, DHT discovery, direct connection). There are no malicious peers.

This PR:
Peer X later connects to peer B and _appends_ new addresses given by peer B for peer A. Peer X dials Peer A using out-of-date addresses that have a higher score first, but eventually uses one of the updated addresses which have no score after some time.

Previous with signed peer record:
Peer X later connects to peer B and _sets_ the new addresses given by peer B for peer A. Peer X can be sure that Peer B is giving the correct address set for peer A and therefore can use that address set (discarding addresses that it knows that peer A is not using). Peer B dials Peer A using the correct address set without having to first discover the previous addresses are out of date.